kolibri/utils/version.py

```diff
@@ -1,106 +1,84 @@
+    return major
 
-        if self.build_number:
-            version = "{version}.dev{build}".format(
-                version=version, build=self.build_number
-            )
 
-        return version
+def get_complete_version(version=None):
+    """Returns a tuple of the version. If version argument is non-empty,
+    then checks for correctness of the tuple provided.
+    """
+    if version is None:
+        from kolibri import VERSION as version
+    else:
+        assert len(version) == 5
+        assert version[3] in ('alpha', 'beta', 'rc', 'final')
 
-    def fullversion(self):
-        """Return both the main version, and the git hash if available."""
+    return version
 
-        mainver = str(self)
 
-        if self.build_hash:
-            mainver = "{mainver}-{build_hash}".format(
-                mainver=mainver, build_hash=self.build_hash
-            )
+def get_docs_version(version=None):
+    version = get_complete_version(version)
+    if version[3] != 'final':
+        return 'dev'
+    else:
+        return '%d.%d' % version[:2]
 
-        return mainver
 
-class ChDir(object):
+def git_describe():
     """
-    Step into a directory temporarily.
-    Taken from:
-    https://pythonadventures.wordpress.com/2013/12/15/chdir-a-context-manager-for-switching-working-directories/
+    :returns: None if no git tag available (no git, no tags, or not in a repo)
     """
-    def __init__(self, path):
-        self.old_dir = os.getcwd()
-        self.new_dir = path
-
-    def __enter__(self):
-        os.chdir(self.new_dir)
-
-    def __exit__(self, *args):
-        os.chdir(self.old_dir)
-
-
-def derive_version_from_git_tag(directory):
     version_string = None
 
-    with ChDir(directory):
-
-        try:
-            with tempfile.TemporaryFile() as discarded_stderr_f:
-                git_describe_string = (
-                    subprocess.check_output(['git', 'describe', '--tags'],
-                                            stderr=discarded_stderr_f
-                                            ).rstrip().decode('utf-8')
-                )  # cast from a byte to a string
-        except (
-            subprocess.CalledProcessError,  # not a git repo
-            OSError  # git executable doesn't exist
-        ):
-            pass
-        else:
-            version_string = parse_git_tag_version_string(git_describe_string)
-
-        return version_string
-
+    try:
+        repo_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        p = subprocess.Popen(
+            'git describe --tags',
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+            shell=True, cwd=repo_dir, universal_newlines=True
+        )
+        version_string = p.communicate()[0].rstrip()
+    except (
+        subprocess.CalledProcessError,  # not a git repo
+        OSError  # git executable doesn't exist
+    ):
+        pass
 
-def derive_version_from_version_file(package):
```

Comment:
This was done to allow these version utils to be used in more than just `kolibri`, e.g. in `kolibri_exercise_perseus_renderer`.

Maybe we should extract these general purpose version utils out into LE utils?